File: pytools/examples/cpp/client/protocol/ReqRoleLogin.cpp
#ifndef _REQ_ROLE_LOGIN_
#define _REQ_ROLE_LOGIN_

#include <list>

#include <pb_type.hpp>
#include <Packet.hpp>

#include <Msg.hpp>



class ReqRoleLogin
{
private:
	U32 m_uid;
	U32 m_uuid;
	U16 m_sid;
	U16 m_cid;
	U32 m_login_time;
	String m_pwd;
	U8 m_relink;
	U8 m_debug;
	String m_os;
	String m_version;


public:
	Packet Encode()
	{
		Packet packet;
		packet.WriteU32(m_uid);
		packet.WriteU32(m_uuid);
		packet.WriteU16(m_sid);
		packet.WriteU16(m_cid);
		packet.WriteU32(m_login_time);
		packet.WriteString(m_pwd);
		packet.WriteU8(m_relink);
		packet.WriteU8(m_debug);
		packet.WriteString(m_os);
		packet.WriteString(m_version);
		packet.Encode(Msg::P_REQ_ROLE_LOGIN);
		return packet;
	}


	void SetUid(U32 uid)
	{
		m_uid = uid;
	}

	void SetUuid(U32 uuid)
	{
		m_uuid = uuid;
	}

	void SetSid(U16 sid)
	{
		m_sid = sid;
	}

	void SetCid(U16 cid)
	{
		m_cid = cid;
	}

	void SetLoginTime(U32 login_time)
	{
		m_login_time = login_time;
	}

	void SetPwd(String pwd)
	{
		m_pwd = pwd;
	}

	void SetRelink(U8 relink)
	{
		m_relink = relink;
	}

	void SetDebug(U8 debug)
	{
		m_debug = debug;
	}

	void SetOs(String os)
	{
		m_os = os;
	}

	void SetVersion(String version)
	{
		m_version = version;
	}

};

#endif

File: pytools/examples/php/client/protocol/ReqRoleLogin.php
<?php
namespace protocol;

use network\Packet;


class ReqRoleLogin
{
	private $uid = 0;
	private $uuid = 0;
	private $sid = 0;
	private $cid = 0;
	private $loginTime = 0;
	private $pwd = '';
	private $relink = 0;
	private $debug = 0;
	private $os = '';
	private $version = '';


	public function encode()
	{
		$packet = new Packet();
		$packet->writeU32($this->uid);
		$packet->writeU32($this->uuid);
		$packet->writeU16($this->sid);
		$packet->writeU16($this->cid);
		$packet->writeU32($this->loginTime);
		$packet->writeString($this->pwd);
		$packet->writeU8($this->relink);
		$packet->writeU8($this->debug);
		$packet->writeString($this->os);
		$packet->writeString($this->version);
		return $packet->encode(Msg::$P_REQ_ROLE_LOGIN);
	}


	public function setUid($uid)
	{
		$this->uid = $uid;
	}

	public function setUuid($uuid)
	{
		$this->uuid = $uuid;
	}

	public function setSid($sid)
	{
		$this->sid = $sid;
	}

	public function setCid($cid)
	{
		$this->cid = $cid;
	}

	public function setLoginTime($loginTime)
	{
		$this->loginTime = $loginTime;
	}

	public function setPwd($pwd)
	{
		$this->pwd = $pwd;
	}

	public function setRelink($relink)
	{
		$this->relink = $relink;
	}

	public function setDebug($debug)
	{
		$this->debug = $debug;
	}

	public function setOs($os)
	{
		$this->os = $os;
	}

	public function setVersion($version)
	{
		$this->version = $version;
	}

}
